erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -326,7 +328,7 @@ frappe.ui.form.on('Payment Entry', {
 							() => {
 								frm.set_party_account_based_on_party = false;
 								if (r.message.bank_account) {
-									frm.set_value("party_bank_account", r.message.bank_account);
+									frm.set_value("bank_account", r.message.bank_account);
```

Comment:
The `get_party_details` method in Payment Entry calls `get_party_bank_account` from `bank_account.py`. There it simply fetches the **default bank account** from the Party mentioned in the Payment Entry. (Check `get_party_bank_account` in `bank_account.py`)
From what i see, the earlier version seems apt.

Reply:
The current implementation in the supplier doctype only allows to link a company bank account:
https://github.com/frappe/erpnext/blob/3c25308b27458413dafaf3094247ec2b1f326d95/erpnext/buying/doctype/supplier/supplier.js#L20-L26

So in other words, a company bank account must be linked in the supplier master, but in Payment Entry it fetches this same bank account for the "party bank account" field.

Maybe I'm missing something, but from what I see one of both has to be wrong (either the query filter in the supplier master or the add_fetch in payment entry). However, @deepeshgarg007 mentioned in https://github.com/frappe/erpnext/pull/20632#pullrequestreview-359493500 that it's supposed to be the company bank account.